packages/build/src/plugins_core/deploy/plugin.js

```diff
@@ -0,0 +1,13 @@
+const {
+  env: { NETLIFY_BUILD_EVENTS_ORDER },
+} = require('process')
+
+const { startBuildbotClient, closeBuildbotClient, deploySiteWithBuildbotClient } = require('./buildbot_client')
+
+const onBuild = async function({ constants: { BUILDBOT_SERVER_SOCKET } }) {
+  const client = await startBuildbotClient(BUILDBOT_SERVER_SOCKET)
+  await deploySiteWithBuildbotClient(client)
+  await closeBuildbotClient(client)
```

Comment:
For cleanup, `connection.end()` should be called if either:
  - `connection.on('error')` is triggered while awaiting `pEvent(buildbotClient, 'connect')`
  - an error is thrown in `deploySiteWithBuildbotClient()`

I.e. we should use `try`/`finally`, and might need to split `startBuildbotClient()` to separate the two initial stages (creating the socket and waiting for the connection).